### ISSUE TITLE
Mel loss computation in the generator

### ISSUE THREAD

**Issue by akashrajkn** [rishikksh20/HiFi-GAN#8]
In the training step, at some point, mel loss is computed: 
```
mel_fake = stft.mel_spectrogram(fake_audio.squeeze(1))
```
`mel_fake` will not have a gradient function. If you check out the `stft.mel_spectrogram` function, you can see this line `magnitudes = magnitudes.data` which basically removes gradient (I guess this was done for better performance?). This results in `loss_mel` not having a gradient function and it won't contribute to training. If you agree, I can submit a PR correcting this. 

Cheers, and really nice that you implemented the paper :)

**Comment by rishikksh20** [comment]
Sure
